### PR TITLE
[MM-30595] Add support for relative icon path to bindings and forms

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
With this PR apps can define `Icon` for both bindings and forms using a relative path.

Please note that no validation of the asset is done due to performance concerns. Invalid icon with show as invalid in the UI. Only malformed path containing e.g. `/../` are logged and ignored.

Tested with hello-world app.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30595

### CODE REVIEW

**Review by levb — APPROVED** [review]
some nits

File: apps/binding.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ package apps
 //              "bindings": [
 //                  {
 //                      "location": "send-button",
-//                      "icon": "http://localhost:8080/static/icon.png",
```
Comment (larkox) [inline apps/binding.go:68]: Should we keep some icons as absolute paths as an example of both?

Comment (larkox) [inline apps/binding.go:68]: Should we create "broken examples" for the sake of testing?

**Review by larkox — APPROVED** [review]
LGTM, just two small considerations.

File: server/proxy/proxy.go [hunk 2]
```diff
@@ -61,13 +64,40 @@ func (p *Proxy) Call(sessionID, actingUserID string, creq *apps.CallRequest) *ap
...
+// If icon is an absolute URL, it's not changed.
+// Otherwise assume it's a path to a static asset and the static path URL prepended.
+func convertToStaticPath(conf config.Config, appID apps.AppID, icon string) (string, error) {
+	if !strings.HasPrefix(icon, "http://") && !strings.HasPrefix(icon, "https://") {
+		cleanIcon, err := utils.CleanStaticPath(icon)
+		if err != nil {
+			return "", errors.Wrap(err, "invalid icon path")
+		}
+
+		icon = conf.StaticURL(appID, cleanIcon)
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/proxy/proxy.go:95]: 2/5 that `StaticURL` should stay in `conf` as it's similar to `AppURL`. Given that I'm inclined to leave normalizeStaticPath as it is.

File: apps/binding.go (hunk 1 above)
Comment (hanzei) [reply apps/binding.go:68]: The example should show the idiomatic way of using things. Hence, the icons should all be relative paths as this is the way we encourage devs to use them.

Comment (hanzei) [reply apps/binding.go:68]: Good point :+1:

**Comment by hanzei** [comment]
Docs PR: https://github.com/mattermost/mattermost-developer-documentation/pull/840

**Comment by hanzei** [comment]
@DHaussermann QA review for a downstream PR was done by @jfrerich in https://github.com/mattermost/mattermost-app-zendesk/pull/60#issuecomment-853968843. Do you still want to test this?

**Review by DHaussermann — APPROVED** [review]
Peer tested as part of a downstream PR

Will re-test for 0.7.0 release testing. 

LGTM!

**Pull request merged**